flink-staging/flink-table/src/main/scala/org/apache/flink/api/java/table/JavaBatchTranslator.scala

```diff
@@ -54,6 +55,22 @@ class JavaBatchTranslator extends PlanTranslator {
     Table(Root(rowDataSet, resultFields))
   }
 
+  override def createTable(tableSource: TableSource): Table = {
+    // a TableSource requires an ExecutionEnvironment
+    if (env.isEmpty) {
+      throw new ExpressionException("This operation requires an ExecutionEnvironment.")
+    }
+    tableSource match {
+      case adaptive: AdaptiveTableSource => Table(Root(adaptive, adaptive.getOutputFields()))
+
+      case static: StaticTableSource =>
+        createTable(static.createStaticDataSet(env.get),
```

Comment:
I think it's more scalaesque to use Scala's pattern matching to extract the `ExecutionEnvironment` from `env` instead of the `if` check with a following `get`.